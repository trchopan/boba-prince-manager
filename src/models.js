export const profileParser = data => ({
  fullname: data.fullname || "",
  position: data.position || "user",
  avatar: data.avatar || "/images/bloody-smile.jpg"
});
