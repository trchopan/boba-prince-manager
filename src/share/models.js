export const themes = {
  light: "Light",
  dark: "Dark"
};

export const languages = {
  vi: { code: "vi", name: "Tiáº¿ng Viá»‡t", char: "ðŸ‡»ðŸ‡³" },
  en: { code: "en", name: "English", char: "ðŸ‡¬ðŸ‡§" }
};

export const profileParser = data => ({
  fullname: data.fullname || "",
  position: data.position || "user",
  avatar: data.avatar || "/images/bloody-smile.jpg"
});
