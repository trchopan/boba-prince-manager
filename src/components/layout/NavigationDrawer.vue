<template>
    <v-navigation-drawer v-model="drawerOpen"
      app
      clipped
      width="200">
      <v-list class="pa-0">
        <v-list-tile v-for="item in drawerItems"
          :key="'drawer-' + item.name"
          avatar
          :to="item.path">
          <v-list-tile-avatar>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-title>{{ item.name }}</v-list-tile-title>
        </v-list-tile>
        <v-divider  />
      </v-list>
    </v-navigation-drawer>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  name: "NavigationDrawer",
  computed: {
    drawerOpen: {
      get: function() {
        return this.$store.state.layout.drawerOpen;
      },
      set: function(value) {
        if (value !== this.$store.state.layout.drawerOpen) {
          this.$store.dispatch("layout/toggleDrawer");
        }
      }
    },
    ...mapGetters({ $t: "layout/$t" }),
    drawerItems() {
      return [
        { path: "/", name: this.$t.drawer.home, icon: "home" },
        {
          path: "/dashboard",
          name: this.$t.drawer.dashboard,
          icon: "dashboard"
        },
        { path: "/about", name: this.$t.drawer.about, icon: "" }
      ];
    }
  }
};
</script>
