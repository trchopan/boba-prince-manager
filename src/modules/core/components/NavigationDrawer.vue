<template>
  <v-navigation-drawer v-model="drawerOpen" app clipped width="230">
    <v-list class="pa-0">
      <v-list-tile
        v-for="item in drawerItems"
        :key="'drawer-' + item.name"
        avatar
        :to="item.path"
      >
        <v-list-tile-avatar>
          <v-icon>{{ item.icon }}</v-icon>
        </v-list-tile-avatar>
        <v-list-tile-title>{{ item.name }}</v-list-tile-title>
      </v-list-tile>
      <v-divider/>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
import { mapGetters } from "vuex";

export const languagesMap = {
  home: { vi: "Trang chủ", en: "Home" },
  dashboard: { vi: "Bảng điều khiển", en: "Dashboard" },
  user: { vi: "Quản lý người dùng", en: "User management" },
  about: { vi: "Về chúng tôi", en: "About" }
};

export const _drawerItems = [
  { path: "/", name: "home", icon: "home" },
  {
    path: "/dashboard",
    name: "dashboard",
    icon: "dashboard"
  },
  { path: "/user", name: "user", icon: "account_circle" },
  { path: "/about", name: "about", icon: "" }
];

export default {
  name: "NavigationDrawer",
  computed: {
    ...mapGetters({
      _drawerOpen: "layout/drawerOpen",
      language: "layout/language"
    }),
    $t() {
      return this.$translate(languagesMap, this.language.code);
    },
    drawerOpen: {
      get: function() {
        return this._drawerOpen;
      },
      set: function(value) {
        if (value !== this.$store.state.layout.drawerOpen) {
          this.$store.dispatch("layout/toggleDrawer");
        }
      }
    },
    drawerItems() {
      return _drawerItems.map(x => ({ ...x, name: this.$t[x.name] }));
    }
  }
};
</script>
