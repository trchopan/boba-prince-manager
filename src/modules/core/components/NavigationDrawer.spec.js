import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Translate from "@/plugins/translate.js";
import NavigationDrawer, { languagesMap } from "./NavigationDrawer.vue";
import { mockCustomElements } from "@/__mocks__/custom-elements.js";
import { languages } from "@/modules/core/layout.models.js";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Translate);

describe("NavigationDrawer.vue", () => {
  let state;
  let getters;
  let actions;
  let store;
  let wrapper;
  let mocDrawerItems = [
    {
      path: "/",
      name: "home",
      icon: "home"
    }
  ];

  beforeEach(() => {
    state = { layout: { drawerOpen: false } };
    getters = {
      "layout/language": () => languages.en,
      "layout/drawerOpen": () => false
    };
    actions = {
      "layout/toggleDrawer": jest.fn()
    };
    store = new Vuex.Store({
      state,
      getters,
      actions
    });
    wrapper = shallowMount(NavigationDrawer, {
      stubs: mockCustomElements,
      store,
      localVue
    });
  });

  it("render drawer items", () => {
    wrapper.vm.items = mocDrawerItems;
    const tile = wrapper.find("v-list-tile-stub");
    const icon = tile.find("v-list-tile-avatar-stub").find("v-icon-stub");
    const title = tile.find("v-list-tile-title-stub");
    expect(tile.attributes("to")).toEqual(mocDrawerItems[0].path);
    expect(icon.text()).toEqual(mocDrawerItems[0].icon);
    expect(title.text()).toEqual(languagesMap.home.en);
  });
  it("toggles drawer", () => {
    wrapper.vm.drawerOpen = true;
    expect(actions["layout/toggleDrawer"]).toBeCalled();
  });
  it("matches snapshot", () => {
    wrapper.vm.items = mocDrawerItems;
    expect(wrapper).toMatchSnapshot();
  });
});
