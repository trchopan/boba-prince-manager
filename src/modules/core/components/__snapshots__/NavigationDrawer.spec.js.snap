// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationDrawer.vue matches snapshot 1`] = `
<v-navigation-drawer-stub app="" clipped="" width="200">
  <v-list-stub class="pa-0">
    <v-list-tile-stub avatar="" to="/">
      <v-list-tile-avatar-stub>
        <v-icon-stub>home</v-icon-stub>
      </v-list-tile-avatar-stub>
      <v-list-tile-title-stub>Home</v-list-tile-title-stub>
    </v-list-tile-stub>
    <v-divider-stub></v-divider-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
